backend/src/main/java/com/ody/mate/service/MateService.java

```diff
@@ -35,4 +33,10 @@ public MateSaveResponse saveAndSendNotifications(MateSaveRequest mateSaveRequest
         notificationService.saveAndSendNotifications(meeting, mate, member.getDeviceToken());
         return MateSaveResponse.from(mate);
     }
+
+    public MeetingFindByMemberResponse findByMeetingAndMember(Meeting meeting, Member member) {
+        int mateCount = mateRepository.countByMeetingId(meeting.getId());
+        Mate mate = mateRepository.findByMeetingIdAndMemberId(meeting.getId(), member.getId());
+        return MeetingFindByMemberResponse.of(meeting, mateCount, mate);
+    }
```

Comment:
[의견]

지금도 잘 읽히지만 mateService에서 MeetingFindBYMemberResponse를 반환하는 흐름은 개인적으로는 어색하게 느껴지는 것 같아요. Meeting 관련 정보가 필요한 MeetingFindByMemberResponse를 mateService가 알고 있기도 하고요!

mateService에서는 meeting_id와 member_id로 mate만 찾아주는 메서드
해당 meeting에 참여한 mateCount가 몇명인지 찾아주는 메서드를 열어주고
domain to dto로 감싸는 나머지 로직은 상위단에서 처리해주는 건 어떨까요? 

다만 이럴 경우에는 mateService에서 mate domain을 반환해야 해서 일관성이 깨져보일 수도 있을 것 같아요.
 ```suggestion
    public Mate findByMeetingAndMember(Meeting meeting, Member member) {
       return  mateRepository.findByMeetingIdAndMemberId(meeting.getId(), member.getId());
    }
    
    public int countMeetingMates(Meeting meeting){
      return mateRepository.countByMeetingId(meeting.getId());
    }
     //나머지 domain to dto는 call한 상위 서비스에서 수행
```

Reply:
꼭 mateRepository -> mateService -> meetingService 흐름을 가져가야 할까요? 차라리 그냥 mateRepository 메서드를 meetingService에서 바로 사용하는 방향은 어떤가요?

```java
public class MeetingService {

    private final MateRepository mateRepository;

    public MeetingFindByMemberResponses findAllByMember(Member member) {
        return meetingRepository.findAllByMember(member)
                .stream()
                .map(meeting -> {
                    int mateCount = mateRepository.countByMeetingId(meeting.getId());
                    Mate mate = mateRepository.findByMeetingIdAndMemberId(meeting.getId(), member.getId());
                    return MeetingFindByMemberResponse.of(meeting, mateCount, mate);
                })
                .sorted(Comparator.comparing(MeetingFindByMemberResponse::date)
                        .thenComparing(MeetingFindByMemberResponse::time))
                .collect(Collectors.collectingAndThen(Collectors.toList(), MeetingFindByMemberResponses::new));
    }
```

Reply:
좋아요! 오히려 그게 더 깔끔할 것 같은데요? 저도 그 방향이 나은 것 같아요

Reply:
[반영] [0fd28b5](https://github.com/woowacourse-teams/2024-ody/pull/251/commits/0fd28b5186047a7eb89d5efbad7ba81a5aaf42d8)